apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderTest.java

```diff
@@ -219,21 +228,87 @@ public void setup() {
     when(metricRegistry.timer(any())).thenReturn(mockTimer);
     when(mockTimer.time()).thenReturn(mockTimerContext);
 
-    // transformer mocking
-    when(carrierClaimTransformer.transform(any(), anyBoolean())).thenReturn(testEob);
-    when(inpatientClaimTransformer.transform(any())).thenReturn(testEob);
-    when(outpatientClaimTransformer.transform(any())).thenReturn(testEob);
-    when(partDEventTransformer.transform(any())).thenReturn(testEob);
-    when(hhaClaimTransformer.transform(any())).thenReturn(testEob);
-    when(snfClaimTransformer.transform(any())).thenReturn(testEob);
-    when(hospiceClaimTransformer.transform(any())).thenReturn(testEob);
-    when(dmeClaimTransformer.transform(any(), anyBoolean())).thenReturn(testEob);
+    // NPI and FDA drug mocking
+    when(mockNpiOrgLookup.retrieveNPIOrgDisplay(Optional.empty())).thenReturn(Optional.empty());
+    when(mockDrugDisplayLookup.retrieveFDADrugCodeDisplay(Optional.empty())).thenReturn("JUNK");
 
     setupLastUpdatedMocks();
 
     mockHeaders();
   }
 
+  /** Sets up claim transformers. */
+  private void setupTransformers() {
+    carrierClaimTransformer =
+        new CarrierClaimTransformerV2(metricRegistry, mockDrugDisplayLookup, mockNpiOrgLookup);
+    partDEventTransformer = new PartDEventTransformerV2(metricRegistry, mockDrugDisplayLookup);
+    dmeClaimTransformer = new DMEClaimTransformerV2(metricRegistry, mockDrugDisplayLookup);
+    hhaClaimTransformer = new HHAClaimTransformerV2(metricRegistry, mockNpiOrgLookup);
+    hospiceClaimTransformer = new HospiceClaimTransformerV2(metricRegistry, mockNpiOrgLookup);
+    inpatientClaimTransformer = new InpatientClaimTransformerV2(metricRegistry, mockNpiOrgLookup);
+    outpatientClaimTransformer =
+        new OutpatientClaimTransformerV2(metricRegistry, mockDrugDisplayLookup, mockNpiOrgLookup);
+    snfClaimTransformer = new SNFClaimTransformerV2(metricRegistry, mockNpiOrgLookup);
```

Comment:
Not sure if you're still in the middle of changes, but these should be mocked since this is the resource provider's unit test. The transformers are out-of-bounds for testing the logic of the resource provider. Tests using the real transformers should be in an IT or E2E.